e2e/e2etests/test_rpc.go

```diff
@@ -0,0 +1,156 @@
+		require.NotEmpty(r, txInfo.Nonce)
+		require.NotEmpty(r, txInfo.Gas)
+
+		blockNumber, err := hexutil.DecodeBig(receipt.BlockNumber)
+		require.NoError(r, err)
+
+		blockByNumber, err := rpcWrapper.EthGetBlockByNumber(r.Ctx, blockNumber, false)
+		require.NoError(r, err)
+		require.NotNil(r, blockByNumber)
+		require.Equal(r, receipt.BlockNumber, blockByNumber.Number)
+		require.NotEmpty(r, blockByNumber.Hash)
+		require.NotEmpty(r, blockByNumber.ParentHash)
+		require.NotEmpty(r, blockByNumber.Timestamp)
+		require.GreaterOrEqual(r, len(blockByNumber.Transactions), 1)
+
+		blockByHash, err := rpcWrapper.EthGetBlockByHash(r.Ctx, common.HexToHash(receipt.BlockHash), false)
+		require.NoError(r, err)
+		require.NotNil(r, blockByHash)
+		require.Equal(r, receipt.BlockHash, blockByHash.Hash)
+		require.NotEmpty(r, blockByHash.Number)
+		require.NotEmpty(r, blockByHash.ParentHash)
+		require.NotEmpty(r, blockByHash.Timestamp)
+		require.GreaterOrEqual(r, len(blockByHash.Transactions), 1)
+
+		traceTx, err := rpcWrapper.DebugTraceTransaction(r.Ctx, common.HexToHash(txHash))
+		require.NoError(r, err)
+		require.NotNil(r, traceTx)
+		require.NotEmpty(r, traceTx.Type)
+		require.NotEmpty(r, traceTx.From)
+		require.NotEmpty(r, traceTx.To)
+
+		traceBlock, err := rpcWrapper.DebugTraceBlockByNumber(r.Ctx, blockNumber)
+		require.NoError(r, err)
+		require.NotNil(r, traceBlock)
+		require.GreaterOrEqual(r, len(traceBlock), 1)
+		require.NotNil(r, traceBlock[0].Result)
+		require.NotEmpty(r, traceBlock[0].Result.From)
+		require.NotEmpty(r, traceBlock[0].Result.To)
+	}
+
+	r.Logger.Info("JSON-RPC tests completed successfully!")
+}
+
+// seedAndDiscoverUserTestTxs creates test transactions and discovers existing transactions
+// from this user via CCTX queries to enable testing RPC methods against historical transactions.
+func seedAndDiscoverUserTestTxs(r *runner.E2ERunner) []string {
+	r.Logger.Info("seeding example transactions")
+	amount := utils.ParseBigInt(r, "10000000000000000")
+
+	// perform the deposit
+	tx := r.ETHDeposit(r.EVMAddress(), amount, gatewayevm.RevertOptions{OnRevertGasLimit: big.NewInt(0)}, true)
+
+	// wait for the cctx to be mined
+	cctx := utils.WaitCctxMinedByInboundHash(r.Ctx, tx.Hash().Hex(), r.CctxClient, r.Logger, r.CctxTimeout)
+	r.Logger.CCTX(*cctx, "deposit")
+	require.Equal(r, crosschaintypes.CctxStatus_OutboundMined, cctx.CctxStatus.Status)
+
+	amount = utils.ParseBigInt(r, "9000000000000000")
+
+	r.ApproveETHZRC20(r.GatewayZEVMAddr)
+
+	// perform the withdraw
+	tx = r.ETHWithdraw(r.EVMAddress(), amount, gatewayzevm.RevertOptions{OnRevertGasLimit: big.NewInt(0)})
+
+	// wait for the cctx to be mined
+	cctx2 := utils.WaitCctxMinedByInboundHash(r.Ctx, tx.Hash().Hex(), r.CctxClient, r.Logger, r.CctxTimeout)
+	r.Logger.CCTX(*cctx2, "withdraw")
+	require.Equal(r, crosschaintypes.CctxStatus_OutboundMined, cctx2.CctxStatus.Status)
+
+	cctxsRes, err := r.CctxClient.CctxAll(r.Ctx, &crosschaintypes.QueryAllCctxRequest{})
```

Comment:
In the live networks, does it mean fetching the 1st page of 100 CCTXs and filtering out those triggered by one of `ZEVMAuth.From.Hex()` and  `EVMAuth.From.Hex()`? I'm not sure how many relevant CCTXs we can get in this case, especially when the CCTX traffic is high in ZetaChain.

Do we really have to filter by address?